### PR TITLE
[LPK][GDI32] Enable mirroring via version resource

### PR DESCRIPTION
**Pull request by peterooch**
## Purpose

Add a mechanism that detects via the version resource if to mirror the application layout or not.
This is a "follow up" to commit 5b14b6f58196fec0dc7c899c26c93a68d4c8542d, even though I think there are better ways to implement mirroring.

This code is taken from the [`GetProcessDefaultLayout`](https://github.com/wine-mirror/wine/blob/377243b41105085ed23407eb2a03b54737d59f64/dlls/user32/win.c#L3994) function from Wine, which is [called](https://github.com/wine-mirror/wine/blob/377243b41105085ed23407eb2a03b54737d59f64/dlls/user32/win.c#L1463) in Wine's internal `WIN_CreateWindowEx` function.

In windows this happens when the GDI is loading the lpk.

Snippet to check if it works in welcome in ReactOS:
https://gist.github.com/peterooch/5e443168af19252036a5b541a4767497

Note:
SetProcessDefaultLayout currently doesn't function correctly.
[Second attempt at fixing it.](https://github.com/peterooch/reactos/commit/cc7f3601dbd869ec04ac9b8180dcdec865ece793)

Includes "dummy code" in `GdiInitializeLanguagePack` which can be of use, or not...

JIRA issue: N/A

### CODE REVIEW

**Comment by HBelusca** [comment]
> In windows this happens when the GDI is loading the lpk.

But is it done *by* LPK, or by GDI just after it has loaded LPK?

**Comment by learn-more** [comment]
@HBelusca `lpk.dll` imports `user32!SetProcessDefaultLayout` so that seems fine.

It does not import version.dll, but it does import:
`ntdll!LdrAccessResource` and `LdrFindResourceEx_U`.
However, I am fine with using the version api's for now.

ps:
I doubt this code should be called from dllmain

**Comment by peterooch** [comment]
The equivalent internal function is called from `LpkInitialize()` which is loaded in `GdiInitializeLanguagePack`, but it's not clear how exactly it does it, it also does several things in addition of calling the mirroring function.

Should I "halfplement" it so it will be loaded in the `LoadLPK` function and it then will call the mirroring function?

File: dll/win32/lpk/lpk.c [hunk 1]
```diff
@@ -109,6 +110,55 @@ static void LPK_DrawUnderscore(HDC hdc, int x, int y, LPCWSTR str, int count, in
...
+{  
+    static const WCHAR translationW[] = { '\\','V','a','r','F','i','l','e','I','n','f','o',
+                                          '\\','T','r','a','n','s','l','a','t','i','o','n', 0 };
+    static const WCHAR filedescW[] = { '\\','S','t','r','i','n','g','F','i','l','e','I','n','f','o',
+                                       '\\','%','0','4','x','%','0','4','x',
+                                       '\\','F','i','l','e','D','e','s','c','r','i','p','t','i','o','n',0 };
+    WCHAR *str, buffer[MAX_PATH];
+#ifdef __GNUC__
+    DWORD i, version_layout = 0;
+    UINT len;
```
[the remark this answers is not in the document]
Comment (learn-more) [reply dll/win32/lpk/lpk.c]: sure

[the remark this answers is not in the document]
Comment (peterooch) [reply dll/win32/lpk/lpk.c]: Should I put `__REACTOS__` ifdef because this a wine code block?

[the remark this answers is not in the document]
Comment (HBelusca) [reply dll/win32/lpk/lpk.c]: Ok then, `#ifdef __REACTOS__`  is indeed better.

**Pull request merged**